pkg/controller/opentelemetrycollector/reconcile_service.go

```diff
@@ -2,19 +2,44 @@ package opentelemetrycollector
 
 import (
 	"context"
+	"errors"
 	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
 
 	"github.com/go-logr/logr"
+	"gopkg.in/yaml.v2"
 	corev1 "k8s.io/api/core/v1"
-	"k8s.io/apimachinery/pkg/api/errors"
+	errs "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/labels"
-	"k8s.io/apimachinery/pkg/util/intstr"
 
 	"github.com/open-telemetry/opentelemetry-operator/pkg/apis/opentelemetry"
 	"github.com/open-telemetry/opentelemetry-operator/pkg/apis/opentelemetry/v1alpha1"
 )
 
+var (
+	// ErrInvalidYAML represents an error in the format of the configuration file
+	ErrInvalidYAML = errors.New("couldn't parse the opentelemetry-collector configuration to assemble a list of ports for the service")
+
+	// ErrNoReceivers indicates that there are no receivers in the configuration
+	ErrNoReceivers = errors.New("no receivers available as part of the configuration")
+
+	// ErrReceiversNotAMap indicates that the receivers property isn't a map of values
+	ErrReceiversNotAMap = errors.New("receivers property in the configuration doesn't contain valid receivers")
+
+	// DNS_LABEL constraints: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-label-names
+	dnsLabelValidation = regexp.MustCompile("^(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])?$")
+)
+
+const (
+	defaultGRPCPort          int32 = 14250
```

Comment:
These are all jaeger specific - shouldn't a more generic table be defined that can also include other receiver's defaults?

Reply:
Possibly a table based on receiver and optional protocol names - if protocols always defined as the node name under the receiver, then could also be used to derive parsing of the receiver's configuration.

Reply:
I'm not a big fan of the code structure as it is right now, there's too much happening at this "reconcile_service" source file. I think I'll do a small refactoring and move it to another package. Might be a good opportunity to move jaeger-specific things to a separate source.

Reply:
Not sure a jaeger specific part is required - if as mentioned [here](https://github.com/open-telemetry/opentelemetry-operator/pull/25#issuecomment-645387944) a "default receiver/protocol port table" is used. So any receiver developers could add their details to the table?

Reply:
The notion of protocol is unique to Jaeger at the moment, so, we'd still need a special treatment.